### PR TITLE
feat: Implement fetching user audiences

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
This PR provides for a new API on the current user: mParticle.Identity.getCurrentUser().getUserAudiences(callback).

The url it accesses is: https://jssdks.mparticle.com/v1/{apikey}/audiences?mpid=3984273492347328 and will return to the user the following schema to a callback:

```
{
  currentAudiences: Audience[],
  pastAudiences: Audience[]
}
```

 ## Testing Plan
 - [] Was this tested locally? If not, explain why.
We are waiting for the backend to be updated before 

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5994

### CODE REVIEW

File: src/audienceManager.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import {
+    FetchUploader,
+    XHRUploader,
+    AsyncUploader,
+    fetchPayload
+} from './batchUploader';
+import Audience from './audience';
+
+type AudienceResponseMessage = 'ar';
+export interface IAudienceServerResponse {
```
Comment (alexs-mparticle) [inline src/audienceManager.ts:11]: We should append the word `Minified` to these interfaces just so we can differentiate from the "Full" interfaces if we ever implement them.

File: src/batchUploader.ts [hunk 2]
```diff
@@ -424,25 +424,26 @@ export class BatchUploader {
     }
 }
 
-abstract class AsyncUploader {
+export abstract class AsyncUploader {
```
Comment (alexs-mparticle) [inline src/batchUploader.ts:427]: If you're reusing the async uploader, let's just move them into their own file.

File: test/src/tests-audience-manager.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        const audienceManager = new AudienceManager(
+            Constants.DefaultBaseUrls.userAudienceUrl,
+            apiKey,
+            newLogger,
+            testMPID
+        );
+
+        expect(audienceManager.logger).to.be.ok;
+        expect(audienceManager.url).to.equal(userAudienceUrl);
+        expect(audienceManager.url).to.equal(userAudienceUrl);
```
Comment (alexs-mparticle) [inline test/src/tests-audience-manager.ts:70]: Is this a duplicate test?

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

File: test/src/tests-audience-manager.ts (hunk 3 above)
Comment (rmi22186) [reply test/src/tests-audience-manager.ts:70]: Good catch

File: src/batchUploader.ts (hunk 2 above)
Comment (alexs-mparticle) [reply src/batchUploader.ts:427]: Let's do it!

File: test/src/tests-audience-manager.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    });
+
+    afterEach(() => {
+        sinon.restore();
+        mockServer.reset();
+        fetchMock.restore();
+    });
+
+    it('should have proper properties on AudienceManager initialization', () => {
+        const newLogger:SDKLoggerApi = new Logger(window.mParticle.config);
```
Comment (alexs-mparticle) [inline test/src/tests-audience-manager.ts:60]:
```suggestion
        const newLogger: SDKLoggerApi = new Logger(window.mParticle.config);
```

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

**Review by alexs-mparticle — APPROVED** [review]

File: src/identity.js [hunk 5]
```diff
@@ -1265,6 +1267,23 @@ export default function Identity(mpInstance) {
             getFirstSeenTime: function() {
                 return mpInstance._Persistence.getFirstSeenTime(mpid);
             },
+            /**
+             * Get user segments for user
+             * @method getUserSegments
+             * @return {Object} an array with user segments????
```
Comment (BrandonStalnaker) [inline src/identity.js:1273]: comments probably need cleaned up here?

**Review by BrandonStalnaker — COMMENTED** [review]
Looks good just 1 minor thing

**Comment by rmi22186** [comment]
@BrandonStalnaker i updated.  please take another look.

File: src/audienceManager.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+}
+export interface IMinifiedAudienceMembership {
+    id: number; // AudienceId
+    n: string; // External Name
+    c: AudienceListMembershipUpdateType[]; // AudienceMembershipChanges
+    s: string[]; // ModuleEndpointExternalReferenceKeys
+}
+
+export enum AudienceMembershipChangeAction {
+    Unknown = 'unkown',
```
Comment (BrandonStalnaker) [inline src/audienceManager.ts:25]: misspelling?

Comment (rmi22186) [reply src/audienceManager.ts]: fixed

**Pull request merged**